src/main/marketplace/inventory/inventory-service.js

```diff
@@ -64,12 +64,14 @@ export class InventoryFetcher {
 export class SelfkeyInventoryFetcher extends InventoryFetcher {
 	async fetch() {
 		let inventory = await this.fetchInventory();
-		const categories = inventory
+		const itemsCategory = inventory
 			.filter(itm => itm.status === 'active')
 			.map(itm => itm.category);
 
+		const categories = [...new Set(itemsCategory)];
+
 		const data = await Promise.all(
-			[...new Set(categories)].map(async category => {
```

Comment:
the set is used to ensure array is unique, check out this change does not break anything

Reply:
Yes, it was moved above that line because it's used again later:
`const categories = [...new Set(itemsCategory)];`